dashboard/lib/services/globally_unique_identifiers.rb

```diff
@@ -71,5 +71,17 @@ def self.find_resource(key)
         find_by(key: keys["CourseVersion"], "course_offerings.key": keys["CourseOffering"])
       return Resource.find_by(key: keys["Resource"], course_version: course_version)
     end
+
+    # Returns a key which can be used by to globally and uniquely identify the
+    # given Resource object
+    #
+    # Note that currently, this is a one-way only process; we do not yet have a
+    # way of finding lessons by this key because we do not yet have a way of
+    # building the regular expression because neither scripts nor lessons have
+    # expression validation for their relevant identifiers.
```

Comment:
I'd say it's a slightly broader issue than that; it's not just that we're using a slash as a delimiter and are worried about slashes showing up in the keys, it's that we don't enforce ANYTHING about the format of the keys, so no matter WHAT we use as a delimiter there's a chance of an issue. Similar to how we had to add validation for vocab, resource, and course version in order to get the special markdown syntaxes to work.

I don't currently anticipate needing to do the inverse operation. If we did end up wanting it, I think it would just be a matter of enforcing formatting for these model values (which is something that would be good to do anyway) and then picking a delimiter that works with that enforced format.